server/datastore/mysql/app_configs.go

```diff
@@ -81,7 +81,13 @@ func (d *Datastore) SaveAppConfig(info *fleet.AppConfig) error {
 		return errors.New("MySQL Event Scheduler must be enabled to configure Host Expiry.")
 	}
 
-	if err := d.ManageHostExpiryEvent(info.HostExpiryEnabled, info.HostExpiryWindow); err != nil {
+	tx, err := d.db.Beginx()
+	if err != nil {
+		return err
+	}
```

Comment:
Rather than manually managing the transaction, please use `withRetryTxx`(as in https://github.com/fleetdm/fleet/blob/c5280c0517b5f184a62f0a1696ea3c0b4b25caa1/server/datastore/mysql/packs.go#L14). I'm open to discussing whether this is the best approach, but I think we should aim for consistency now.

Reply:
+1 for consistency. But should this be retried? Maybe we should add a `withTx` func instead? That would make it cleaner for the rollback at the very least.

Reply:
I am on board with adding a `withTx` function.